pkg/routeagent/controllers/route/route.go

```diff
@@ -165,6 +258,153 @@ func (r *Controller) processNextCluster() bool {
+	ipSlice := strings.Split(ipAddr, ".")
+	if len(ipSlice) < 4 {
+		return nil, fmt.Errorf("Invalid ipAddr [%s]", ipAddr)
+	}
+
+	ipSlice[0] = strconv.Itoa(VxLANVTepNetworkPrefix)
+	vxlanIP := net.ParseIP(strings.Join(ipSlice, "."))
+	return vxlanIP, nil
+}
+
+func (r *Controller) getHostIfaceIPAddress() (net.IP, *net.IPNet, error) {
+	addrs, err := r.defaultHostIface.Addrs()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if len(addrs) > 0 {
+		for i := range addrs {
+			ipAddr, ipNetwork, err := net.ParseCIDR(addrs[i].String())
+			if err != nil {
+				klog.Errorf("Unable to ParseCIDR : %v\n", addrs)
+			}
+			if ipAddr.To4() != nil {
+				return ipAddr, ipNetwork, nil
+			}
+		}
+	}
+	return nil, nil, err
+}
+
+func (r *Controller) createVxLANInterface(ifaceType int, gatewayNodeIP net.IP) error {
+	ipAddr, vtepMask, err := r.getHostIfaceIPAddress()
+	if err != nil {
+		return fmt.Errorf("Unable to retrieve the IPv4 address on the Host %v", err)
+	}
+
+	vtepIP, err := r.getVxlanVtepIPAddress(ipAddr.String())
+	if err != nil {
+		return fmt.Errorf("Failed to derive the vxlan vtepIP %v", err)
+	}
+
+	if ifaceType == VxInterfaceGateway {
+		attrs := &vxLanAttributes{
+			name:     VxLANIface,
+			vxlanId:  100,
+			group:    nil,
+			srcAddr:  nil,
+			vtepPort: VxLANPort,
+			mtu:      1450,
+		}
+
+		r.vxlanDevice, err = newVxlanIface(attrs)
+		if err != nil {
+			return fmt.Errorf("Failed to create vxlan interface on Gateway Node: %v", err)
+		}
+
+		for fdbAddress := range r.remoteVTEPs.Set {
+			err = r.vxlanDevice.AddFDB(net.ParseIP(fdbAddress), "00:00:00:00:00:00")
+			if err != nil {
+				return fmt.Errorf("Failed to add FDB entry on the Gateway Node vxlan iface %v", err)
+			}
+		}
+
+		// Enable loose mode (rp_filter=2) reverse path filtering on the vxlan interface.
+		err = ioutil.WriteFile("/proc/sys/net/ipv4/conf/"+VxLANIface+"/rp_filter", []byte("2"), 0644)
+		if err != nil {
+			return fmt.Errorf("Unable to update vxlan rp_filter proc entry, err: %s", err)
+		} else {
+			klog.Info("Successfully configured rp_filter to loose mode(2) ")
+		}
+
+	} else if ifaceType == VxInterfaceWorker {
+		// non-Gateway/Worker Node
+		attrs := &vxLanAttributes{
+			name:     VxLANIface,
+			vxlanId:  100,
+			group:    gatewayNodeIP,
+			srcAddr:  vtepIP,
+			vtepPort: VxLANPort,
+			mtu:      1450,
+		}
+
+		r.vxlanDevice, err = newVxlanIface(attrs)
+		if err != nil {
+			return fmt.Errorf("Failed to create vxlan interface on non-Gateway Node: %v", err)
+		}
+	}
+
+	err = r.vxlanDevice.configureIPAddress(vtepIP, vtepMask.Mask)
+	if err != nil {
+		return fmt.Errorf("Failed to configure vxlan interface ipaddress on the Gateway Node %v", err)
+	}
+
+	return nil
+}
+
+func (r *Controller) processNextPod() bool {
+	obj, shutdown := r.podWorkqueue.Get()
+	if shutdown {
+		return false
+	}
+	err := func() error {
+		defer r.podWorkqueue.Done(obj)
+
+		key := obj.(string)
+		ns, name, err := cache.SplitMetaNamespaceKey(key)
+		if err != nil {
```

Comment:
You need to call `r.podWorkqueue.Forget(obj)` here and all return paths below that don't re-queue.